fix some buttons are not responding after theme change

Parveshdhull opened the pull request:
fixes: https://github.com/status-im/status-mobile/issues/14780

### Issue Cause
- Chat was not closing on system theme change
- As `:close-chat` and `:navigate-to-chat` both had `:dispatch` key, it was causing `:merging-fx-with-common-keys` error, due to unclosed chat
- Models were not dismissed on theme change

status: ready

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 8cc5592e | [#1](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-14811/1/) | 2023-01-17 19:25:30 | ~2 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-14811/1/consoleText) |

Parveshdhull commented:
maybe we should also improve [safe-merge](https://github.com/status-im/status-mobile/blob/9a60fc160070f76f9e02062344d07b616876a0ea/src/utils/re_frame.cljs#L42) function, so that it will convert `dispatch` into `dispatch-n`, instead of throwing error `merging-fx-with-common-keys`

VladimrLitvinenko commented:
Hi @Parveshdhull could you resolve the conflicts, please?

flexsurfer commented on src/status_im/events.cljs:
```diff
@@ -14,6 +14,7 @@
    status-im.chat.models.loading
    status-im.chat.models.transport
    [status-im.constants :as constants]
```
oh thought we removed status-im.constants

flexsurfer commented:
yeah this one is tricky one, just wondering what if won't reset root, in new code it should work, just use hot-reload trick instead

Parveshdhull commented:
> yeah this one is tricky one, just wondering what if won't reset root, in new code it should work, just use hot-reload trick instead

we can try, but not sure if it will work: https://wix.github.io/react-native-navigation/docs/style-theme#changing-theme-dynamically

flexsurfer commented:
> > yeah this one is tricky one, just wondering what if won't reset root, in new code it should work, just use hot-reload trick instead
> 
> we can try, but not sure if it will work: https://wix.github.io/react-native-navigation/docs/style-theme#changing-theme-dynamically

we don't use RNN themes, so it won't work, but hot reload might work, we just need to force re-render entire react tree

Parveshdhull commented:
@VladimrLitvinenko, thank you for picking up PR. Please can you pause testing, I am thinking of adding few more changes to this PR. I will ping you once PR will be ready for testing.

VladimrLitvinenko commented:
> @VladimrLitvinenko, thank you for picking up PR. Please can you pause testing, I am thinking of adding few more changes to this PR. I will ping you once PR will be ready for testing.

Sure. Let me know when it is ready

status-im-auto commented:
## 0% of end-end tests have passed
```
Total executed tests: 10
Failed tests: 10
Passed tests: 0
Not executed tests: 15
```
```
IDs of not executed tests: 702730,702782,702731,702745,702813,702855,702733,702783,702784,702774,702775,702776,702807,702808,702846 
```
```
IDs of failed tests: 702840,702732,702850,702851,702845,702841,702839,702842,702838,702843 
```
<h3>Not executed tests (15)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14811.apk&PR_ID=14811&TR_CASE_IDS=702730,702782,702731,702745,702813,702855,702733,702783,702784,702774,702775,702776,702807,702808,702846">Rerun not executed tests</a></li></details><h3>Failed tests (10)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14811.apk&PR_ID=14811&TR_CASE_IDS=702840,702732,702850,702851,702845,702841,702839,702842,702838,702843">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3077433">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3077437">test_community_leave</a>, id: 702845 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3077434">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3077432">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3077435">test_community_mark_all_messages_as_read</a>, id: 702842 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3077431">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Button` by `xpath`: `//*[@text="Next"]` </div><div>Device 2: Tap on found: Button </div></blockquote></p>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[blocked by #14797]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/203dafa2b360486998b1bae10f18e688?auth=8e96f2097b1d8617acd4d8b454b9a624">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/203dafa2b360486998b1bae10f18e688/0009screenshot.png?auth=8e96f2097b1d8617acd4d8b454b9a624">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fbf4463a83ba4b378239f957342badb3?auth=8945a373583869fedd2fcbd424cf00f5">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/fbf4463a83ba4b378239f957342badb3/0009screenshot.png?auth=8945a373583869fedd2fcbd424cf00f5">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3077436">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[blocked by #14797]]</b><br/><br/></td></tr></tbody></table><h4>Class TestGroupChatMediumMultipleDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3077419">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `xpath`: `//*[@text="Next"]` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
Test setup failed: critical/chats/test_group_chat.py:289: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[blocked by #14637]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ea985cfd480647fba99e40b0d16cec6c?auth=6cb17f5e57ad8e175030c4d26ac80b55">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/ea985cfd480647fba99e40b0d16cec6c/0009screenshot.png?auth=6cb17f5e57ad8e175030c4d26ac80b55">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d4fc79cb9a1b4369983bca3284b17948?auth=af0a5244688266493b17352795da8c6a">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/d4fc79cb9a1b4369983bca3284b17948/0009screenshot.png?auth=af0a5244688266493b17352795da8c6a">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3077439">test_activity_center_decline_contact_request_no_pn</a>, id: 702850 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `xpath`: `//*[@text="Next"]` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
Test setup failed: medium/test_activity_center.py:192: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/187ea094bfd9480cb83a622268ea459f?auth=37b7045179bb3521bfbfe836d2deccc0">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/187ea094bfd9480cb83a622268ea459f/0009screenshot.png?auth=37b7045179bb3521bfbfe836d2deccc0">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a012d2a929b346bdbf61f3dbe6c27c8d?auth=82b804c73160c962826bd31785af392c">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/a012d2a929b346bdbf61f3dbe6c27c8d/0009screenshot.png?auth=82b804c73160c962826bd31785af392c">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3077440">test_activity_center_mentions_in_community_jump_to</a>, id: 702851 </b></td></tr><tr><td>

```
Test setup failed: medium/test_activity_center.py:192: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[blocked by #14798]]</b><br/><br/></td></tr></tbody></table></details>

Parveshdhull commented:
Hi @VladimrLitvinenko, Thank you for your patience. PR is ready to be tested again.

Apart from #14780, please also test the behavior of status bar icon colors, and nav bar colors as per system theme changes and also app theme changes.
PR also remove the new-ui-toggle as it's not functional anymore, So please also test for regression for account creation and login.

### Known issue
These changes might create [this](https://github.com/status-im/status-mobile/issues/12921) distorted app state on old screens while changing theme. It's fine for now, we need to re-implement these screens without the RNN theme. But if distortion happening on new screens, please let me know.

flexsurfer reviewed (approved):
nice

qoqobolo commented:
Hi @Parveshdhull , thanks for the fixes!
There is an issue with the status bar icon colors on both Android and iOS, take a look, please.

### ISSUE 1: Black font color of the status bar icons

**Case 1:**
1. Have your device in dark mode
2. Install Status 

In this case, the status bar icons have a black font with a black background from the beginning.

<img width="326" alt="Screenshot 2023-01-23 at 12 35 38" src="https://user-images.githubusercontent.com/67952253/214030111-def70220-ee9d-43d4-ba8e-211d6b259d9d.png">

**Case 2:**
1. Have your device in light mode
2. Install Status and create an account

You can see on the video, that in this case, the status bar looks okay during onboarding, but on the first `Jump to` screen icons get black with a black background.
The status bar gets also dark on other screens if you change the app theme to dark.

https://user-images.githubusercontent.com/67952253/214029504-23541d86-21a6-45d8-be90-9eb3cbc791fb.mp4


On 14 sec you can also see that the `Appearance` title stays light when changing theme, but as I understood it may be a part of this known issue:

> Known issue
These changes might create https://github.com/status-im/status-mobile/issues/12921 distorted app state on old screens while changing theme. It's fine for now, we need to re-implement these screens without the RNN theme. But if distortion happening on new screens, please let me know.

Parveshdhull commented:
hi @qoqobolo, Thank you for testing the PR and finding this issue. Issue should be fixed now.

qoqobolo commented:
Thanks @Parveshdhull !

### ISSUE 2: The status bar icons get white on the white background (or vice versa) if switch the mode while a community channel is open

**Steps:**
1. Switch the theme to dark on your device
2. Install Status
3. Create a community
4. Open a channel
5. Switch the mode to light on your device while the channel is opened
**Actual result:** status bar gets white with white icons


https://user-images.githubusercontent.com/67952253/214089880-7558f654-8811-4010-ac29-cb0b73022429.mp4

---
Also, ISSUE 1 (Case 2) is still reproducible on iOS.

**Steps:**
1. Have your device in light mode
2. Install Status and go through the onboarding flow until the first `Jump to` screen
**Actual result:** fully black status bar


<img width="390" alt="Screenshot 2023-01-23 at 17 15 24" src="https://user-images.githubusercontent.com/67952253/214090941-0abea0b2-36b4-4337-8405-3a096fececf1.png">

Parveshdhull commented:
Hi @qoqobolo,

**Issue 1:** changing the status bar color without the re-initializing of app, is currently not working in ios.
We can also reproduce this behavior by opening and closing the shell screen(press jump-to).
Status bar icon color will remain the same in the both home-stack and shell screen.
Somewhat related issue: https://github.com/wix/react-native-navigation/issues/5446

**Issue 2:** Currently we are not storing the ids of all opened screens, so on theme change we only change the status-bar icon color for the root screen. But once those screens closed and reopened the status bar color should be correct.

As Issue 1 looks a little not fixable(needs further research) and Issue 2 non blocker, I think we can skip them for now and log them separately. wdyt?

Parveshdhull commented:
So, for ios, we are not able to change the status-bar icon color at runtime, but I made changes so that on login it should use the correct color.
**Before Login Screens:** Same as a system theme
**After Login Screens:** Same as app theme on most screens (except shell screen)

Please let me know if anything is not clear, It's a little confusing with so many edge cases. And thanks again for testing PR.

status-im-auto commented:
## 0% of end-end tests have passed
```
Total executed tests: 10
Failed tests: 10
Passed tests: 0
Not executed tests: 15
```
```
IDs of not executed tests: 702730,702782,702731,702745,702813,702855,702733,702783,702784,702774,702775,702776,702807,702808,702846 
```
```
IDs of failed tests: 702840,702732,702850,702851,702845,702841,702839,702842,702838,702843 
```
<h3>Not executed tests (15)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-230123-211222-2896fe-pr14811-x86.apk&PR_ID=14811&TR_CASE_IDS=702730,702782,702731,702745,702813,702855,702733,702783,702784,702774,702775,702776,702807,702808,702846">Rerun not executed tests</a></li></details><h3>Failed tests (10)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-230123-211222-2896fe-pr14811-x86.apk&PR_ID=14811&TR_CASE_IDS=702840,702732,702850,702851,702845,702841,702839,702842,702838,702843">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3078743">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3078747">test_community_leave</a>, id: 702845 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3078744">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3078742">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3078745">test_community_mark_all_messages_as_read</a>, id: 702842 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3078741">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `xpath`: `//*[@text="Next"]` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[blocked by #14797]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d2a44a3b3979402685b2b07915456a58?auth=0cc12c6d27750d04a470df315743b727">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/d2a44a3b3979402685b2b07915456a58/0009screenshot.png?auth=0cc12c6d27750d04a470df315743b727">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3cce098b013440af889edc98a5d18292?auth=dcdfcc71537f52f0258f1916b1f01dd1">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/3cce098b013440af889edc98a5d18292/0009screenshot.png?auth=dcdfcc71537f52f0258f1916b1f01dd1">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3078746">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td>

```
Test setup failed: critical/test_public_chat_browsing.py:586: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[blocked by #14797]]</b><br/><br/></td></tr></tbody></table><h4>Class TestGroupChatMediumMultipleDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3078729">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `xpath`: `//*[@text="Next"]` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
Test setup failed: critical/chats/test_group_chat.py:289: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[blocked by #14637]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5d2d962608624b00a46fdaa36648e9cb?auth=2611f8c775cbe0c4a9a11cb8f802fe6f">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/5d2d962608624b00a46fdaa36648e9cb/0009screenshot.png?auth=2611f8c775cbe0c4a9a11cb8f802fe6f">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/646da08cfeb14edbbd444f49b2611bc5?auth=e116d57eb5788d917b3e7711a6632c60">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/646da08cfeb14edbbd444f49b2611bc5/0009screenshot.png?auth=e116d57eb5788d917b3e7711a6632c60">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3078749">test_activity_center_decline_contact_request_no_pn</a>, id: 702850 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Button` by `xpath`: `//*[@text="Next"]` </div><div>Device 2: Tap on found: Button </div></blockquote></p>

```
Test setup failed: medium/test_activity_center.py:192: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3cc65ec02654496da4b4b4fcd0daccbc?auth=ec6a6327dffa0882a984ddebf43d6f9d">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/3cc65ec02654496da4b4b4fcd0daccbc/0009screenshot.png?auth=ec6a6327dffa0882a984ddebf43d6f9d">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9b24284812124ed2a601fa7488edc878?auth=e7cd773e587746cb07955f78fb40cc4e">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/9b24284812124ed2a601fa7488edc878/0009screenshot.png?auth=e7cd773e587746cb07955f78fb40cc4e">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3078750">test_activity_center_mentions_in_community_jump_to</a>, id: 702851 </b></td></tr><tr><td>

```
Test setup failed: medium/test_activity_center.py:192: in prepare_devices
    self.loop.run_until_complete(run_in_parallel(((self.device_1.create_user,), (self.device_2.create_user,))))
/usr/lib/python3.10/asyncio/base_events.py:646: in run_until_complete
    return future.result()
__init__.py:35: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:204: in create_user
    self.chats_tab.wait_for_visibility_of_element(30)
../views/base_element.py:134: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: ChatsTab by accessibility id:`chats-stack-tab` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[blocked by #14798]]</b><br/><br/></td></tr></tbody></table></details>

qoqobolo commented:
> As Issue 1 looks a little not fixable(needs further research) and Issue 2 non blocker, I think we can skip them for now and log them separately. wdyt?

Sure, I'll log them in one new low-prio issue just to track them.
Thanks for the explanation! Overall, everything looks good to me.

But one more thing: e2e are failing because the `Enable Notifications` and `Welcome to Status` screens are back. Looks like they accidentally snuck into the code again :) Or it's on purpose? Can you clarify please?

https://user-images.githubusercontent.com/67952253/214276370-7dc9721f-052e-46ad-a156-5651f01896c7.mp4

Parveshdhull commented:
> Looks like they accidentally snuck into the code again :) Or it's on purpose? Can you clarify please?

I also removed new-ui-toggle in PR and that accidentally exposed the else part(welcome screen).
Removed it, thank you for noticing.

qoqobolo commented:
Thanks @Parveshdhull !
I don't see any other issues besides the ones mentioned.
PR can be merged.

Parveshdhull committed:
fix some buttons are not responding after theme change
